monai/utils/jupyter_utils.py

```diff
@@ -0,0 +1,351 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This set of utility function is meant to make using Jupyter notebooks easier with MONAI. Plotting functions using 
+Matplotlib produce common plots for metrics and images. 
+"""
+
+from enum import Enum
+from threading import RLock, Thread
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+
+# from monai.utils import exact_version, optional_import
+
+# if TYPE_CHECKING:
+#     import matplotlib.pyplot as plt
+#     from ignite.engine import Engine, Events
+
+#     Figure = plt.Figure
+#     Axes = plt.Axes
+#     has_matplotlib = True
+# else:
+#     Engine, _ = optional_import("ignite.engine", "0.4.4", exact_version, "Engine")
+#     Events, _ = optional_import("ignite.engine", "0.4.4", exact_version, "Events")
+#     plt, has_matplotlib = optional_import("matplotlib.pyplot")
+#     Figure, _ = optional_import("matplotlib.pyplot", name="Figure")
+#     Axes, _ = optional_import("matplotlib.pyplot", name="Axes")
+
+try:
+    import matplotlib.pyplot as plt
+
+    has_matplotlib = True
+except ImportError:
+    has_matplotlib = False
+
+try:
+    from ignite.engine import Engine, Events
+
+    has_ignite = True
+except ImportError:
+    Engine = object
+    Events = object
+    has_ignite = False
+
+LOSS_NAME = "loss"
+
+
+def plot_metric_graph(
+    ax,
+    title: str,
+    graphmap: Dict[str, Union[List[float], Tuple[List[float], List[float]]]],
+    yscale: str = "log",
+    avg_keys: Tuple[str] = (LOSS_NAME,),
+    window_fraction: int = 20,
+):
+    """
+    Plot metrics on a single graph with running averages plotted for selected keys. The values in `graphmap`
+    should be lists of (timepoint, value) pairs as stored in MetricLogger objects.
```

Comment:
could you create a few tests for the plotting method and MetricLogger? came across this util not sure how reliable it is https://github.com/matplotlib/matplotlib/blob/6fe34e72d2b7daf8586a08a8611cc73b3c0ad60c/lib/matplotlib/testing/decorators.py#L393

Reply:
I forgot to request changes for unit tests, could you please have a follow-up PR, thank you! @ericspod